### DISCUSSION TITLE
Is `PandasAI` (class) for backward compatibility purposes only?

### DISCUSSION THREAD

**Discussion by mspronesti** [sinaptik-ai/pandas-ai#503] [General]
Hi @gventuri,
I noticed that, while from v1.0 onwards the whole logic seems to have been moved to `SmartDataframe`, `PandasAI` (class) is still supported. Is this for backward compatibility purposes only?

It seems like that quite some people are still using pandasai "the old way" (ie. with `PandasAI`) facing some troubles as not all the features have been maintained in the transition process. Before opening a PR for them, I'd like to make sure these considerations are correct.

Also, if it is the case, I believe a deprecation warning shown when instantiating a `PandasAI` class would make sense.

**Comment by gventuri** [comment]
@mspronesti that's a very good point. Yes, it's indeed only maintained for backward compatibility and will be removed completely in the future. The effort has been to support for most of the previous cases but at the same time not supporting for additional new features not to incentivize using it.

One thing we could do every time it is instantiated is logging some kind of warning, letting the user know that it will be deprecated and redirecting them to the docs. What do you think?

**Reply by mspronesti** [reply]
@gventuri Yep, I believe we should display a warning, e.g. 

 ```py
warnings.warn("The class `PandasAI` is deprecated from version 1.0 and will removed in no longer than X months. Please use `pandasai.SmartDataframe` instead.")
```

However, for the time being I believe we should guarantee a complete backward compatibility, including
- auto-conversion of langchain's llms
- conversationality
- access to attributes (prompts, last generated code etc.)

What do you think?
